.circleci/config.yml

```diff
@@ -615,7 +615,7 @@ jobs:
             testForDebugging="<< parameters.test >>"
 
             if [ -z "$testForDebugging" ]; then
-              detected_changes=$(git diff origin/HEAD --name-only --diff-filter=AM | (grep 'src/test/regress/sql/.*.sql\|src/test/regress/spec/.*.spec' || true))
+              detected_changes=$(git diff origin/main --name-only --diff-filter=AM | (grep 'src/test/regress/sql/.*.sql\|src/test/regress/spec/.*.spec' || true))
```

Comment:
If the current feature branch does not contain all the branches that are in `origin/main` this `git diff` returns more items than needed. I suggest you use a different syntax that checks for file changes with the common ancestor with `origin/main` and not the current reference of `origin/main`

```suggestion
              detected_changes=$(git diff origin/main... --name-only --diff-filter=AM | (grep 'src/test/regress/sql/.*.sql\|src/test/regress/spec/.*.spec' || true))
```

For example, now I checked out `ahmet-testing-failure-flaky` branch that updated only a single file `failure_create_reference_table.sql`. We see many items in `git diff` output because this branch was created before #6568 was merged into main.

```bash
sh$ git diff origin/main --name-only
src/test/regress/expected/upgrade_partition_constraints_after.out
src/test/regress/expected/upgrade_partition_constraints_after_0.out
src/test/regress/expected/upgrade_partition_constraints_before.out
src/test/regress/expected/upgrade_partition_constraints_before_0.out
src/test/regress/expected/upgrade_pg_dist_object_test_after.out
src/test/regress/expected/upgrade_pg_dist_object_test_after_0.out
src/test/regress/expected/upgrade_pg_dist_object_test_after_1.out
src/test/regress/expected/upgrade_pg_dist_object_test_before.out
src/test/regress/expected/upgrade_pg_dist_object_test_before_0.out
src/test/regress/sql/failure_create_reference_table.sql
src/test/regress/sql/upgrade_partition_constraints_after.sql
src/test/regress/sql/upgrade_partition_constraints_before.sql
src/test/regress/sql/upgrade_pg_dist_object_test_after.sql
src/test/regress/sql/upgrade_pg_dist_object_test_before.sql

sh$ git diff origin/main... --name-only
src/test/regress/sql/failure_create_reference_table.sql
```